lib.js

```diff
@@ -0,0 +1,215 @@
+// TODO
+// - decide where to update window.localstorage.{lastCid,ipns,lastPublishedCid}
+// - make function names consistent nounVerb or verbNoun
+
+
+export async function publish(cid){
+  let r = await fetch('ipns://ipmb', {
+    method: 'POST', 
+    body: cid,
+  });
+  const ipns = await r.text();
+  const updateHtml = !window.localStorage.ipns;
+  window.localStorage.ipns = ipns;
+  window.localStorage.lastPublishedCid = cid;
+  return ipns;
+}
+
+export function parseFilename(filename){
+  let date = filename.slice(0, 10);
+  let title = decodeURIComponent(filename.slice(11)).replace(/\.md$/, '');
+  return { date, title };
+}
+
+export function filenameToTitle(filename){
+  return decodeURIComponent(filename.slice(11)).replace(/\.md$/, '');
+}
+
+function parseMeta(file){
+  return {
+    ...file,
+    title: filenameToTitle(file.filename),
+    date: file.filename.slice(0, 10),
+    excerpt: file.content.slice(0, Math.min(file.content.indexOf('\n'), 100)),
+  }
+}
+
+
+// Create markdown blog based on data stored in lastCid
+export async function thisIsBlog(doFileStuff){
+
+  let lastCid = window.localStorage.getItem('lastCid');
+  const previousCid = lastCid;
+ 
+  // make the update
+  lastCid = await doFileStuff();
+
+  // get content and update index
+  let files = await _fetchFolder(lastCid);
+  files = files.map(parseMeta);
+  let indexBody = '';
+  for (let post of files){
+    // this should probably be a relative link
+    indexBody = indexBody.concat(`- **[${post.title}](/ipmb-db/${post.filename})** (posted ${post.date}) - ${post.excerpt}\n`)
```

Comment:
```suggestion
    indexBody += `- **[${post.title}](/ipmb-db/${post.filename})** (posted ${post.date}) - ${post.excerpt}\n`
```

Generally `+=` is more readable for string stuff.

Reply:
Thanks, I'll update. MDN also [recommends](https://developer.mozilla.org/en-US/docs/web/javascript/reference/global_objects/string/concat#performance) += for performance reasons. I've always been avoiding + and += for strings to avoid javascript's tomfoolery with type coercion :)